### PR TITLE
Upcoming version V0.6.1

### PR DESCRIPTION
**Pull request by pedrocamargo**

### CODE REVIEW

**Comment by janzill** [comment]
@pedrocamargo I don't have any pyqt experience so I won't be of much help with that but I'll work my way through it nevertheless. However, given the sheer size of this it might take a couple of days.

**Comment by pedrocamargo** [comment]
Take your time!!!!

Pedro Camargo, Ph.D.
www.xl-optim.com
<http://www.xl-optim.com>

File: docs/source/SiouxFalls.rst [hunk 1]
```diff
@@ -0,0 +1,501 @@
...
+going through centroids (all nodes are centroids), but that is generally not the
+case. For zones with a single connector per zone it is slightly faster to also
+deselect this option, but use this carefully.
+
+.. image:: images/path_computation_configuration.png
+    :width: 282
+    :align: center
+    :alt: path_computation_configuration
+
+If we select that paths need to be in a separate layer, than every time you
```
Comment (janzill) [inline docs/source/SiouxFalls.rst]: ..., then (not than)

File: docs/source/SiouxFalls.rst [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+--------
+
+We can also skim the network to look into general connectivity of the network
+
+.. image:: images/skimming_menu.png
+    :width: 507
+    :align: center
+    :alt: skimming_menu
+
+To perform skim, we can select to compute a matrix from all nodes to all nodes,
```
Comment (janzill) [inline docs/source/SiouxFalls.rst]: skimming instead of skim?

File: docs/source/SiouxFalls.rst [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+The last input is the gravity model itself, which can be done by loading a
+model that has been previously calibrated, or by selecting the deterrence
+function from the drop-down menu and typing the corresponding parameter values.
+
+.. image:: images/apply_gravity_configure_model.png
+    :width: 526
+    :align: center
+    :alt: apply_gravity_configure_model
+
+To run the procedure, simply queue the job (and select the output wile with the
```
Comment (janzill) [inline docs/source/SiouxFalls.rst]: I just looked up 'wile' and google says 'devious or cunning stratagems employed in manipulating or persuading someone to do what one wants'. That's not what you meant, right? :D

File: docs/source/index.rst [hunk 4]
```diff
@@ -1,41 +1,48 @@
...
+   tutorials_and_sample_data
    other
-   softwaredevelopment
+   SiouxFalls
    engine
 
 
+Quick Tour
+----------
+AequilibraE 0.6 for QGIS has brought substantial to the previous graphical
```
Comment (janzill) [inline docs/source/index.rst]: substantial *changes* to ...?

File: docs/source/networkpreparation.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+Direction
+---------
+
+Links need to have directionality associated to them because **links can be**
+**bi-directional** in AequilibraE, which is not the case for several of the
+commercial platforms in the market. For this reason, one needs to have a field
+for link direction with values in the set [-1, 0, 1], where:
+
+* If **direction = -1**, then we the link allows **BA flow only**
```
Comment (janzill) [inline docs/source/networkpreparation.rst]: three 'we' to many I think

File: docs/source/other.rst [hunk 6]
```diff
@@ -5,10 +5,35 @@ Other topics
...
 Contributors
 ~~~~~~~~~~~~
-pass
+
+The list of contributors is always growing, so you can check the lists on the
+`QGIS plugin repository <https://github.com/AequilibraE/AequilibraE-GUI/graphs/contributors>`_
+and the
+`Python library repository. <https://github.com/AequilibraE/aequilibrae/graphs/contributors>`_
+
+As of February/2009, the main contributors are
```
Comment (janzill) [inline docs/source/other.rst]: Thanks for the credit, but I joined way later than 2009 :D

File: docs/source/project.rst [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+AequilibraE Project
+===================
+
+.. toctree::
+   :maxdepth: 2
+
+The AequilibraE project is one of the newest portions of the
+`aequilibrae API <www.aequilibrae.com/python>`_, and therefore not very mature.
+
+For a better over.overview of the AequilibraE project, please check the
```
Comment (janzill) [inline docs/source/project.rst]: Is this (over.overview) on purpose?

File: docs/source/project.rst [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+One would notice that nowhere in the **GUI** one can indicate which modes they
+want to see the network connected for or how to control how many connectors per
+mode will be created.  Although it could be implemented, such a solution would
+be convoluted and there is probably no good reason to do so.
+
+Instead, we have chosen to develop the procedure with the following criteria:
+
+* All modes will be connected to links where those modes are allowed.
+* When considering number of connectors per centroid, there is no guarantee that
+  each and every mode will have that number of connectors. If a particular mode
```
Comment (janzill) [inline docs/source/project.rst:139]: particular mode *is* ...

File: docs/source/other.rst (hunk 6 above)
Comment (pedrocamargo) [reply docs/source/other.rst]: The funny part is that the project started in December 2014... :-D

File: paths_procedures/show_shortest_path_dialog.py [hunk 9]
```diff
@@ -143,44 +166,47 @@ def produces_path(self):
...
+                )
 
-                    # add a feature
-                    all_links = []
-                    for k in self.res.path:
-                        fet = self.link_features[k]
-                        all_links.append(fet)
+    def create_path_with_selection(self):
+        f = 'link_id'
+        t = " or ".join([f"{f}={k}" for k in self.res.path])
```
Comment (janzill) [inline paths_procedures/show_shortest_path_dialog.py:182]: much nicer :)

File: paths_procedures/traffic_assignment_dialog.py [hunk 10]
```diff
@@ -143,118 +326,57 @@ def choose_folder_for_outputs(self):
...
+            if algorithm != 'all-or-nothing':
+                for q in [self.progressbar1, self.progress_label1]:
+                    q.setVisible(True)
+                self.progressbar1.setRange(0, self.miter)
+
+            for q in [self.progressbar0, self.progress_label0]:
+                q.setVisible(True)
+                self.progressbar0.setRange(0, self.project.network.count_centroids())
+
+            if True:
```
Comment (janzill) [inline paths_procedures/traffic_assignment_dialog.py]: is there a reason this is in here?

File: paths_procedures/traffic_assignment_dialog.py [hunk 11]
```diff
@@ -264,334 +386,138 @@ def check_data(self):
...
+        elif val[0] == "rgap":
+            self.rgap = val[1]
         elif val[0] == "finished_threaded_procedure":
             self.job_finished_from_thread()
+        self.progress_label1.setText(f'{self.iter}/{self.miter} - Rel. Gap {self.rgap:.2E}')
 
-    # TODO: Write code to export skims
+    # Save link flows to disk
     def produce_all_outputs(self):
+        fn = os.path.join(self.output_path, "skims.aem")
```
Comment (janzill) [inline paths_procedures/traffic_assignment_dialog.py:412]: I don't think it's used anywhere but below in the else statement, could move to there to make it consistent with the rest of the code maybe

Comment (janzill) [reply paths_procedures/traffic_assignment_dialog.py:412]: wait, I was too early, I underestimated the size of this file, I found it below :)

Comment (janzill) [reply paths_procedures/traffic_assignment_dialog.py:412]: although my point is still valid I guess, the omx filename is constructed twice, could do for this one as well

**Review by janzill — APPROVED** [review]

File: paths_procedures/traffic_assignment_dialog.py (hunk 10 above)
Comment (pedrocamargo) [reply paths_procedures/traffic_assignment_dialog.py]: I have no idea... probably some debugging I was doing at the time...

File: distribution_procedures/distribution_models_dialog.py [hunk 12]
```diff
@@ -251,7 +235,7 @@ def change_vector_field(self, cob_orig, cob_dest, dt):
             if dt == "data":
                 for f in self.datasets[d].fields:
                     if np.issubdtype(self.datasets[d].data[f].dtype, np.integer) or np.issubdtype(
-                        self.datasets[d].data[f].dtype, np.float
+                            self.datasets[d].data[f].dtype, np.float
```
Comment (ychtw) [inline distribution_procedures/distribution_models_dialog.py]: Looks like extra indentation was added here by accident

Comment (pedrocamargo) [reply distribution_procedures/distribution_models_dialog.py]: It is black formatting

**Pull request merged**